nixos/modules/programs/firejail.nix

```diff
@@ -78,12 +98,30 @@ in {
         not wrapped if they specify the absolute path to the binary.
       '';
     };
+
+    wrappedPackages = mkOption {
+      type = with types; listOf package;
+      default = [ ];
+      example = literalExample ''
+        [ pkgs.mpv ]
+      '';
+      description = ''
+        Put a package into <option>systemPackages</option>,
+        but wrap its binaries with firejail.
+        Compared to <option>wrappedBinaries</option>,
+        this e.g. has the advantage of providing desktop entries and icons.
+        However, you should be careful about using these packages'
+        libraries as they will not be wrapped. Note also that applications may
+        not be firejailed if invoked via a desktop file that specifies an
+        absolute path to the unwrapped binary.
```

Comment:
How about adding a check such that if a desktop file contains `Exec=/` then the build fails ?

Reply:
Oh actually such a check exists. I'd fail the build instead of printing a warning, this way there is no "risk" for the user.

Reply:
Agreed.

Reply:
I just ran across a couple issues which I think solidify the position that hard coded store paths in desktop files should be regarded as bugs: #106996, #101859.